secure-app-model/keyvault/cpvsample/src/main/java/com/microsoft/store/samples/secureappmodel/cpvsample/security/KeyVaultProvider.java

```diff
@@ -63,89 +57,32 @@ public String getSecret(String secretName)
-     * @throws InterruptedException {@link InterruptedException}
-     * @throws MalformedURLException {@link MalformedURLException}
-     */
-    private AuthenticationResult getAccessToken(String authorization, String resource, String clientId, String clientSecret) 
-        throws ExecutionException, InterruptedException, MalformedURLException
-    {
-        AuthenticationContext authContext; 
-        AuthenticationResult authResult;
-        ExecutorService service = null;
-        Future<AuthenticationResult> future;
-
-        try
-        {
-            service = Executors.newFixedThreadPool(1);
-            authContext = new AuthenticationContext(authorization, true, service);
-
-            future = authContext.acquireToken(
-                resource,
-                new ClientCredential(
-                    clientId, 
-                    clientSecret), 
-                null);
-            
-            authResult = future.get();
-
-            return authResult;
-        }
-        finally
-        {
-            service.shutdown();
-        }
+        client.setSecret(secretName, value);
     }
 
     /**
      * Gets a client that is capable of interacting with the Azure Key Vault service.
      *
      * @param clientId The identifier of the client requesting the token.
      * @param clientSecret The secure secret of the client requesting the token. 
+     * @param tenantId The tenant Id of the client requesting the token. 
      * 
      * @return A client that is capable of interacting with the Azure Key Vault service.
      */
-    private KeyVaultClientCustom getKeyVaultClient(String clientId, String clientSecret)
+    private SecretClient getKeyVaultClient(String clientId, String clientSecret, String tenantId)
     {
-        return new KeyVaultClient(new KeyVaultCredentials() 
-        {
-            /**
-             * @param authorization Address of the authority to issue the token.
-             * @param resource Identifier of the target resource that is the recipient of the requested token, a URL.
-             * @param scope The scope of the authentication request.
-             *
-             * @return Access token to be used with Azure Key Vault operations.
-             */
-            @Override
-             public String doAuthenticate(String authorization, String resource, String scope) 
-             {
-                 AuthenticationResult authResult; 
-                 
-                 try 
-                 {
-                     authResult = getAccessToken(authorization, resource, clientId, clientSecret);
- 
-                     return authResult.getAccessToken();
-                 }
-                 catch(Exception ex)
-                 {
-                     ex.printStackTrace();
-                 }
- 
-                 return "";
-             }
-        });
+        ClientSecretCredential clientSecretCredential=new ClientSecretCredentialBuilder()
```

Comment:
We want people to use DefaultAzureCredential. It is a chained credential that also encompasses ClientSecretCredential. This method doesn't need to exist. We could roll this up into:

```java
var credential = new DefaultAzureCredentialBulder().build();
var client = new SecretClientBuilder()
    .vaultUrl(vaultBaseUrl)
    .credential(credential)
    .build();
```

The environment variables we set should be. When the DefaultAzureCredential is built, then it will probe for the environment variables to create the credential.
    * AZURE_CLIENT_ID
    * AZURE_CLIENT_SECRET
    * AZURE_TENANT_ID

Here is this blog: https://devblogs.microsoft.com/azure-sdk/authentication-and-the-azure-sdk/